### PR TITLE
Fix bug where delete run path would delete parent directory

### LINKED ISSUE #6740: Delete runpath deletes (way) too much
**Describe the bug**
For given runpaths, delete_runpath will delete folders it should not delete.

**To reproduce**
Steps to reproduce the behaviour:
1. Set RUNPATH to (only) `realization-<IENS>` in poly.ert
2. `ert gui my_config.ert`
3. Run ensemble experiment
4. Run ensemble experiment once more, ticking off to delete the runpath
5. Observe that the entire `test-data` directory in the ert repository has been deleted (and ert will crash)

**Expected behaviour**
Only the specified runpath directories to be deleted

**Environment**
 - ERT release: 8.x
 - Remote/HPC execution involved: no

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #6740 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tests/unit_tests/run_models/test_base_run_model.py [hunk 1]
```diff
@@ -120,3 +124,34 @@ def get_run_path_mock(realizations, iteration=None):
...
+    model_config = ModelConfig(runpath_format_string=run_path_format)
+    subs_list = SubstitutionList()
+    config = MagicMock()
+    config.model_config = model_config
+    config.substitution_list = subs_list
+
+    brm = BaseRunModel(simulation_arguments, config, None, None, None, None)
+    brm.rm_run_path()
+    assert not run_path.exists()
+    assert run_path.parent.exists()
```
Comment (berland) [inline tests/unit_tests/run_models/test_base_run_model.py:157]: Perhaps add an assert that any extra file/dir unknown to ERT alongside `realizations-*` is untouched? It is e.g. customary to have a `share` directory like this.

File: tests/unit_tests/run_models/test_base_run_model.py [hunk 2]
```diff
@@ -109,14 +113,61 @@ def test_check_if_runpath_exists(
...
+        target_case=None,
+        start_iteration=0,
+        iter_num=0,
+        minimum_required_realizations=0,
+        ensemble_size=1,
+        stop_long_running=False,
+    )
+    expected_remaining = []
+    expected_removed = []
+    for iens, mask in enumerate(active_realizations):
```
Comment (oyvindeide) [inline tests/unit_tests/run_models/test_base_run_model.py:150]: Not sure I like this logic, could fix it by putting everything in one parameterize, but then there would be a lot of repetition there, what do you think?

Comment (berland) [reply tests/unit_tests/run_models/test_base_run_model.py:150]: fine with me.

**Review by berland — APPROVED** [review]
LGTM. Seems not possible any longer to get ERT to issue an `rm -rf /` ..

**Comment by berland** [comment]
Ah, that removal is only present in a branch.

**Comment by oyvindeide** [comment]
> Ah, that removal is only present in a branch.

Yes, figured I would back port this to reinstate it 👍

**Pull request merged**